rpmio/expression.c

```diff
@@ -671,6 +682,52 @@ static Value doLogical(ParseState state)
   return NULL;
 }
 
+static Value doTernary(ParseState state)
+{
+  Value v1 = NULL, v2 = NULL;
+  DEBUG(printf("doTernary()\n"));
+
+  v1 = doLogical(state);
+  if (v1 == NULL)
+    goto err;
+  if (state->nextToken == TOK_TERNARY_COND) {
+    int result;
+    switch (v1->type) {
+      case VALUE_TYPE_INTEGER:
+	result = v1->data.i != 0;
+	break;
+      case VALUE_TYPE_STRING:
+	result = v1->data.s[0] != '\0';
+	break;
+      default:
+	goto err;
+    }
+    valueFree(v1);
+    if (rdToken(state))
+      goto err;
+    v1 = doTernary(state);
+    if (v1 == NULL)
+      goto err;
+    if (state->nextToken != TOK_TERNARY_ALT) {
+      exprErr(&state, _("syntax error in expression"), state->p);
```

Comment:
expression.c: In function ‘doTernary’:
expression.c:712:15: warning: passing argument 1 of ‘exprErr’ from incompatible pointer type [-Wincompatible-pointer-types]
  712 |       exprErr(&state, _("syntax error in expression"), state->p);
      |               ^~~~~~

Reply:
Damn. I'm copying from the wrong line ;)